Sources/Data/models/SessionWindow.swift

```diff
@@ -63,6 +63,22 @@ extension SessionWindow {
     return Self.from(windowId: windowId, in: DataController.viewContext)
   }
   
+  public static func createIfNeeded(index: Int32, isPrivate: Bool, isSelected: Bool) {
+    DataController.performOnMainContext { context in
+      if SessionWindow.getActiveWindow(context: context) != nil {
+        return
+      }
+      
+      _ = SessionWindow(context: context, index: index, isPrivate: isPrivate, isSelected: isSelected)
+      
+      do {
+        try context.save()
```

Comment:
doesnt `performOnMainContext` call context.save()?

Reply:
It does not. It does:
```swift
DataController.perform(.existing(MainContext)) {
}
```

Which does:
```swift
case .existing(let existingContext):
     task(existingContext)
```

Only background or new context does:
```swift
case .new(let inMemory):
   try backgroundContext.save()
```

Reply:
Yeah correct. It's always existing, so it never does a `save`. It only does `task(existingContext)`